### PR TITLE
nushell: update to 0.34.0

### PR DESCRIPTION
**Pull request by cinerea0**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by cinerea0** [comment]
Persistent CI errors seem to be due to this issue: https://github.com/rust-lang/crates.io/issues/3782. It may not be resolved until a cargo update gets released.

File: srcpkgs/nushell/template [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 pkgname=nushell
-version=0.33.0
+version=0.34.0
 revision=1
 build_style=cargo
 configure_args="--features=extra"
 hostmakedepends="pkg-config python3 libgit2-devel"
-makedepends="openssl-devel libgit2-devel libX11-devel libxcb-devel"
+makedepends="openssl-devel libgit2-devel libX11-devel libxcb-devel
+ libatomic-devel"
```
Comment (ericonr) [inline srcpkgs/nushell/template:9]: Shouldn't this be added only for the `XBPS_TARGET_NO_ATOMIC8` case?

Comment (cinerea0) [reply srcpkgs/nushell/template]: The most recent push failed on i686 despite that arch having ATOMIC8, so I think it should be for all arch's.

Comment (ericonr) [reply srcpkgs/nushell/template:9]: Did it fail because of network issues or with `couldn't find __atomic*` errors?

Comment (cinerea0) [reply srcpkgs/nushell/template:9]: The latter. I don't have a screenshot so I can't prove it, but it said it couldn't find `atomic*`.

Comment (ericonr) [reply srcpkgs/nushell/template:9]: Just mention it in the commit message, then :)

`add libatomic for all targets, since build system always uses it`

**Comment by ericonr** [comment]
I fixed the commit message because it was too long ;)

**Comment by cinerea0** [comment]
Thank you!

**Pull request merged**